### PR TITLE
Launcher tweaks

### PR DESCRIPTION
**Pull request by mrnoname1000**
Forgot to PR some improvements to the launcher! Should be pretty self-explanatory and I made some notes going back over them:

Finally got rid of the realpath function from Stackoverflow and replaced it with some more self-explanatory functions. Should be faster now too since it runs as few external commands as possible. Hopefully it doesn't break things, if you want to test the readlink logic you can comment out lines 290-292 or remove realpath from your PATH.

Also I added a couple lines for compatibility with ksh93! I think every Bourne shell that's installable on modern operating systems can launch jruby now.

### CODE REVIEW

**Comment by enebo** [comment]
The carrot of this PR is really that we should stop getting "I can't run this script in my zorronexusobscure linux distro?".  The speed is always good but likely a drop in the bucket compared to actually boostrapping the JVM.  @headius will you review this?  I think we should land it quicker to find potential problems than later.

**Comment by headius** [comment]
Sounds great! I will review and merge this weekend or early next week.

File: bin/jruby.sh [hunk 1]
```diff
@@ -44,70 +52,66 @@ esceval()
...
 # to the named by LISTNAME1, preserving order.
 preextend() {
-    map prepend "$@"
+    eval "$1=\"\${$2} \${$1}\""
+}
+
+# echo [STRING...]
+#
+# Dumb echo, i.e. print arguments joined by spaces with no further processing
+echo() {
```
Comment (headius) [inline bin/jruby.sh:113]: Functions like this will only be visible within this script, correct?

**Comment by headius** [comment]
Ok looks good to me with my level of shell knowledge! I will merge.

**Review by headius — APPROVED** [review]

File: bin/jruby.sh (hunk 1 above)
Comment (headius) [reply bin/jruby.sh:113]: After chatting with @mrnoname1000 and trying some subprocess checks, it appears the functions defined in this launcher script do not propagate beyond the script:

```
$ jruby -e 'system "which esceval"'
$ jruby -e 'system "command -v esceval"'
$
```

Comment (mrnoname1000) [reply bin/jruby.sh:113]: Yeah, unless it's run in bash and exported with `export -f echo` or a preceding `set -a` it won't inherit via the environment.

**Pull request merged**